### PR TITLE
mingw: split windows and linux recipes

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **mingw-w64/***

there is no change at all in the recipe, just a separation of the 2 paths (linux and windows), which are 99% of the time different.

related to #8307

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — APPROVED** [review]

File: recipes/mingw-w64/linux/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+
+class MingwConan(ConanFile):
+    name = "mingw-w64"
+    description = "MinGW is a contraction of Minimalist GNU for Windows"
```
Comment (prince-chrismc) [inline recipes/mingw-w64/linux/conanfile.py:8]: Config and validate say linux 😖 


```suggestion
    description = "MinGW is a contraction of Minimalist GNU for Linux"
```

Comment (ericLemanissier) [reply recipes/mingw-w64/linux/conanfile.py:8]: You want to invent MinGL ? I think you might be on something here !

File: recipes/mingw-w64/config.yml [hunk 2]
```diff
@@ -2,5 +2,5 @@ versions:
     "8.1":
       folder: "all"
     "8.0.2":
-      folder: "all"
+      folder: "linux"
```
Comment (prince-chrismc) [inline recipes/mingw-w64/config.yml:5]: This confuses me

Comment (ericLemanissier) [reply recipes/mingw-w64/config.yml:5]: I am not sure what confuses you, so what follows may not clear up anything, but I'll try anyway: The `linux` folder contains a recipe which is able to build MinGW toolchain on Linux. This recipe compiles a single version for now.
Mingw is a toolchain which always targets Windows (afaik) but can be compiled either om Linux or on windows.

**Comment by ericLemanissier** [comment]
Context: this recipe is more or less "dead" because there is no plan to put newer version on http://sourceforge.net/projects/mingw-w64/. I will make a new windows recipe based on https://github.com/niXman/mingw-builds when a new tag will be made (https://github.com/niXman/mingw-builds/issues/575)

File: recipes/mingw-w64/linux/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+                "--with-mpc={}".format(self.package_folder)
+            ]
+            # with_gmp_mpfc_mpc = [
+            #    "--with-gmp=system",
+            #    "--with-gmp-prefix={}".format(self.deps_cpp_info["gmp"].rootpath.replace("\\", "/")),
+            #    "--with-mpfr=system",
+            #    "--with-mpfr-prefix={}".format(self.deps_cpp_info["mpfr"].rootpath.replace("\\", "/")),
+            #    "--with-mpc=system",
+            #    "--with-mpc-prefix={}".format(self.deps_cpp_info["mpc"].rootpath.replace("\\", "/"))
+            # ]
```
[the remark this answers is not in the document]
Comment (ericLemanissier) [reply recipes/mingw-w64/linux/conanfile.py:130]: @Pro You're the author, are you ok if we remove these lines ?

File: recipes/mingw-w64/config.yml (hunk 2 above)
Comment (prince-chrismc) [reply recipes/mingw-w64/config.yml:5]: That's the magic piece I was missing.

You can target windows from linux!

**Review by prince-chrismc — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by dvirtz** [comment]
@ericLemanissier didn't you intend to remove the linux stuff from the `all` folder? now the linux code is duplicated.

**Comment by ericLemanissier** [comment]
My current plan is to transfer the Windows part to mingw-builds package https://github.com/conan-io/conan-center-index/pull/8349
Then we can fully retire the recipe in mingw/all

**Pull request merged**